Cauli/CauliURLProtocol/CauliURLProtocol.swift

```diff
@@ -48,8 +48,29 @@ internal class CauliURLProtocol: URLProtocol {
     }
 
     override init(request: URLRequest, cachedResponse: CachedURLResponse?, client: URLProtocolClient?) {
-        record = Record(request)
-        super.init(request: request, cachedResponse: cachedResponse, client: client)
+        var modifiedRequest = request
+        let interceptingStream: InterceptingInputStream?
+        if let bodyStream = modifiedRequest.httpBodyStream {
+            interceptingStream = InterceptingInputStream(parentStream: bodyStream)
+            modifiedRequest.httpBodyStream = interceptingStream
+        } else {
+            interceptingStream = nil
+        }
+        
+        record = Record(modifiedRequest)
+        super.init(request: modifiedRequest, cachedResponse: cachedResponse, client: client)
+        interceptingStream?.interceptingDelegate = self
+    }
+}
+
+extension CauliURLProtocol: InterceptingInputStreamDelegate {
+    func didRead(_ buffer: UnsafeMutablePointer<UInt8>, maxLength len: Int) {
+        let data = Data(bytes: buffer, count: len)
```

Comment:
For the response we added a method `append` to the record. I think we should do the same here to be consistent.

Reply:
Works for me. Will change it this way.